spec.bs

```diff
@@ -1856,55 +1875,95 @@ To <dfn>convert to an AuctionAd sequence</dfn> given a [=list=]-or-null |ads|:
-    |allTrustedScoringSignals|["`renderURLs`"][|renderURL|].
-  1. If |adComponentRenderURLs| is not [=list/empty=]:
-    1. Let |adComponentRenderURLsValue| be a new empty [=map=].
-    1. If |allTrustedScoringSignals|["`adComponentRenderURLs`"] [=map/exists=], [=set/for each=]
-      |adComponentRenderURL| in |adComponentRenderURLs|:
-      1. If |allTrustedScoringSignals|["`adComponentRenderURLs`"][|adComponentRenderURL|]
-        [=map/exists=], then [=map/set=] |adComponentRenderURLsValue|[|adComponentRenderURL|] to
-        |allTrustedScoringSignals|["`adComponentRenderURLs`"][|adComponentRenderURL|].
-    1. [=map/Set=] |trustedScoringSignals|["`adComponentRenderURLs`"] to |adComponentRenderURLsValue|.
+1. Let |fullSignalsUrl| be null.
+1. If |auctionConfig|'s [=auction config/trusted scoring signals url=] is not null:
+  1. Let |fullSignalsUrl| be the result of [=building trusted scoring signals url=] with |auctionConfig|'s
+    [=auction config/trusted scoring signals url=], «|renderURL|», |adComponentRenderURLs|,
+    |auctionConfig|'s [=auction config/seller experiment group id=], and |topLevelOrigin|.
+
+    Implementations may batch requests by collecting render URLs and ad component render URLs
+    from multiple invocations of [=score and rank a bid=] and passing them all to a single invocation
+    of [=building trusted scoring signals url=] to get a |scoringSignalsUrl|. By employing this approach,
+    the [=string/length=] of [=URL serializer|serialized=] |scoringSignalsUrl| must not exceed the
+    [=auction config/max trusted scoring signals url length=] of the auction. In cases where the length
+    limit is exceeded, the request must be divided into smaller pieces to comply with the length restriction.
+    However, note that a single request is always considered valid, regardless of whether its length exceeds
+    the auction's [=auction config/max trusted scoring signals url length=].
+
+    The network response has to be parsed to pull out the pieces relevant to each
+    [=evaluating a scoring script|evaluation of a scoring script=].
+
+    These requests may also begin before the script fetch, but requests cross-origin to the
+    script origin must not happen until [:Ad-Auction-Allow-Trusted-Scoring-Signals-From:] header on
+    the script is received, parsed, and determined to authorize such a fetch.
+  1. If |fullSignalsUrl| is not same [=same origin=] with |auctionConfig|'s
+     [=auction config/seller=], then:
+      1. Set |crossOriginTrustedScoringSignalsOrigin| to |fullSignalsUrl|'s [=url/origin=].
+      1. If |allowCrossOriginTrustedScoringSignalsFrom| does not [=list/contain=]
+        |crossOriginTrustedScoringSignalsOrigin|:
+        1. Set |crossOriginTrustedScoringSignalsOrigin| to null.
+        1. Set |fullSignalsUrl| to null.
+1. If |fullSignalsUrl| is not null:
+  1. Let |trustedScoringSignals| and |allTrustedScoringSignals| be null.

```

Comment:
seems no need to declare |trustedScoringSignals| here anymore, just declare it at the line where you set it to a new map (current line 1927).

Reply:
Done